core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -662,6 +662,7 @@ private[spark] class TaskSetManager(
 
     val failureReason = s"Lost task ${info.id} in stage ${taskSet.id} (TID $tid, ${info.host}): " +
       reason.asInstanceOf[TaskFailedReason].toErrorString
+    var failureException: Option[Throwable] = None
     reason match {
```

Comment:
@markhamstra Thanks for the comment. I had another look at made the change you suggested.